Tempo/PlayerNavigationController.swift

```diff
@@ -8,25 +8,78 @@
 
 import UIKit
 
-class PlayerNavigationController: UINavigationController {
+let PostLikedStatusChangeNotification = "PostLikedStatusChange"
+
+protocol PostDelegate {
+	var currentPost: Post? { get }
+}
+
+class PlayerNavigationController: UINavigationController, PostDelegate {
 
 	var playerCell: PlayerCellView!
 	let frameHeight = CGFloat(72)
 	
+	var expandedCell: ExpandedPlayerView!
+	let expandedHeight = CGFloat(198)
+	
+	var postsRef: [Post]?
+	var postRefIndex: Int?
+	var currentPost: Post? {
+		didSet {
+			if let newPost = currentPost {
+				playerCell.updateCellInfo(newPost)
+				expandedCell.updateCellInfo(newPost)
+			}
+		}
+	}
+	
 	override func viewDidLoad() {
 		super.viewDidLoad()
-		let playerFrame = UIView(frame: CGRectMake(0, UIScreen.mainScreen().bounds.height - 72, UIScreen.mainScreen().bounds.width, frameHeight))
+		let playerFrame = UIView(frame: CGRectMake(0, UIScreen.mainScreen().bounds.height - frameHeight, UIScreen.mainScreen().bounds.width, frameHeight))
 		playerFrame.backgroundColor = UIColor.redColor()
-		self.view.addSubview(playerFrame)
-		playerCell = NSBundle.mainBundle().loadNibNamed("PlayerCellView", owner: self, options: nil).first as! PlayerCellView
-		playerCell.setup()
-		playerCell.frame = CGRectMake(0, 0, UIScreen.mainScreen().bounds.width, frameHeight)
-		playerFrame.addSubview(playerCell)
+		view.addSubview(playerFrame)
+		playerCell = NSBundle.mainBundle().loadNibNamed("PlayerCellView", owner: self, options: nil).first as? PlayerCellView
+		playerCell?.setup(self)
+		playerCell?.frame = CGRectMake(0, 0, UIScreen.mainScreen().bounds.width, frameHeight)
+		playerCell?.userInteractionEnabled = false
+		playerFrame.addSubview(playerCell!)
 		// Do any additional setup after loading the view.
+		
+		// Setup expandedCell
+		expandedCell = NSBundle.mainBundle().loadNibNamed("ExpandedPlayerView", owner: self, options: nil).first as? ExpandedPlayerView
+		expandedCell?.setup(self)
+		expandedCell?.frame = CGRectMake(0, UIScreen.mainScreen().bounds.height, UIScreen.mainScreen().bounds.width, expandedHeight)
+		view.addSubview(expandedCell!)
+		
+		
+		NSNotificationCenter.defaultCenter().addObserverForName(PlayerDidFinishPlayingNotification, object: nil, queue: nil) { [weak self] note in
```

Comment:
Please for the love of all that is good and holy, be careful as all hell with the curse that is NSNotificationCenter's block-based APIs!!! (Honestly I just avoid them for the following reasons)

When you call `-addObserverForName:object:queue:usingBlock:` this returns an opaque observer object. If you don't assign this return value to something, you have forever lost this reference to an observer that is trapped inside NSNotificationCenter (until your app inevitably dies from notification overload). Step 1 is to save the result of this call into an ivar (likely an ivar on the class that calls this API). Step 2 is to always make sure you have weak self referencing inside the block (which you have done correctly), otherwise you'll have a retain cycle. Step 3 is to remove the observer you created when calling this API from NSNotificationCenter!!! You should be balancing out your calls to addObserver with removeObserver. Every time you call addObserver, you should have already called removeObserver for the existing observer object (if you don't, you're piling up observers in NSNotificationCenter). One reason I prefer the normal API is because the removal of the observer happens automatically through weak referencing on dealloc in later versions of iOS and macOS. Regardless, it won't help you in cases where you keep calling addObserver multiple times for the lifetime of the object without balancing those calls out with removeObject.